### PR TITLE
Use better action to setup rubies

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
# Description:

Use an action that includes more rubies, so the we don't need to use `rvm` for ruby 2.3 and jruby.

# Tasks:

- [x] Describe the problem / feature
- [ ] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by MSP-Greg** [comment]
@deivid-rodriguez 

Not causing the failures, but actions/checkout@v1 has been updated to

actions/checkout@v2

There were some updates, especially for busy repos...

File: .github/workflows/ubuntu-rvm.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       CLASSPATH: ""
     strategy:
       matrix:
-        ruby: [ '2.3.8', 'ruby-head', 'jruby-9.2.9.0' ]
+        ruby: [ 'ruby-head' ]
     steps:
       - uses: actions/checkout@v1
```
Comment (MSP-Greg) [inline .github/workflows/ubuntu-rvm.yml]: actions/checkout@v2

File: .github/workflows/ubuntu.yml [hunk 2]
```diff
@@ -13,18 +13,13 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby: [ '2.4.x', '2.5.x', '2.6.x' ]
+        ruby: [ '2.3.8', '2.4.9', '2.5.7', '2.6.5', 'jruby-9.2.9.0' ]
         test_tool: [ "rubygems", "bundler" ]
     steps:
       - uses: actions/checkout@v1
```
Comment (MSP-Greg) [inline .github/workflows/ubuntu.yml]: actions/checkout@v2

File: .github/workflows/ubuntu.yml [hunk 3]
```diff
@@ -13,18 +13,13 @@ jobs:
...
       - run: git submodule update -i
-      - name: Fixed world writable dirs
-        run: |
-          chmod go-w $HOME
-          sudo chmod -R go-w /usr/share
-          sudo chmod -R go-w /opt
       - name: Setup ruby
-        uses: actions/setup-ruby@v1
+        uses: actions/use-ruby-action@master
```
Comment (MSP-Greg) [inline .github/workflows/ubuntu.yml]: eregon/use-ruby-action@master

File: .github/workflows/ubuntu.yml [hunk 4]
```diff
@@ -13,18 +13,13 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby: [ '2.4.x', '2.5.x', '2.6.x' ]
+        ruby: [ '2.3.8', '2.4.9', '2.5.7', '2.6.5', 'jruby-9.2.9.0' ]
```
Comment (MSP-Greg) [inline .github/workflows/ubuntu.yml]: Version don't need be quoted, and will default to the latest available, hence:
```
ruby: [ 2.3, 2.4, 2.5, 2.6, 2.7, jruby ]
```

**Comment by deivid-rodriguez** [comment]
I addressed most of @MSP-Greg's feedback, and this PR should be equivalent to the current CI situation, just with a simpler setup. So merging!

@bundlerbot merge

**Pull request merged**